### PR TITLE
Add checks for immutable keys

### PR DESCRIPTION
**Pull request by mhess-swl**
Part of #8466.

This fixes one test in the crypto transfer suite, testing various attempts at using immutable accounts.

### CODE REVIEW

File: hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenCreateHandleParityTest.java [hunk 1]
```diff
@@ -137,7 +137,7 @@ void tokenCreateMissingTreasuryKey() throws PreCheckException {
         final var txn = txnFrom(TOKEN_CREATE_WITH_MISSING_TREASURY);
 
         final var context = new FakePreHandleContext(accountStore, txn);
-        assertThrowsPreCheck(() -> subject.preHandle(context), INVALID_TREASURY_ACCOUNT_FOR_TOKEN);
+        assertThrowsPreCheck(() -> subject.preHandle(context), INVALID_ACCOUNT_ID);
```
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenCreateHandleParityTest.java:140]: Looking at `SigRequirementsTest`, the missing treasury test [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/sigs/order/SigRequirementsTest.java#L3588) shows that `INVALID_TREASURY_ACCOUNT_FOR_TOKEN` is not the correct error code. This test was ported to `TokenCreateHandleParityTest` incorrectly, and is corrected here.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java [hunk 2]
```diff
@@ -319,6 +318,15 @@ private void addCustomFeeCollectorKeys(
         for (final var customFee : customFeesList) {
             final var collector = customFee.feeCollectorAccountIdOrElse(AccountID.DEFAULT);
 
+            // Verify that the collector either has a non-empty alias OR a mutable key
+            final var acctStore = context.createStore(ReadableAccountStore.class);
+            final var collectorAcct = acctStore.getAccountById(collector);
+            if (collectorAcct != null
```
Comment (tinker-michaelj) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java:324]: If this is necessary, it suggests we need to circle back to how we are adding hollow account signing requirements. 🤔 But probably better to do that in a dedicated issue.

Comment (mhess-swl) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java:324]: Sure, I'll open a new issue. For this I followed the mono code as best I could, but there could well be a better way to work the signing requirements.

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java:324]: Agree with @tinker-michaelj . There were some mono-service enhancements happening when implementing this. It is possible we missed some signing requirements for hollow accounts.

Comment (mhess-swl) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java:324]: Opened #9195 to address this

**Pull request merged**